cmake/CMakeLists.txt

```diff
@@ -458,36 +505,31 @@ if(ENABLE_GPU)
     if(NOT BIN2C)
       message(FATAL_ERROR "Couldn't find bin2c, use -DBIN2C helping cmake to find it.")
     endif()
-    include_directories(${CUDA_INCLUDE_DIRS})
-    list(APPEND LAMMPS_LINK_LIBS ${CUDA_LIBRARIES} ${CUDA_CUDA_LIBRARY})
     set(GPU_PREC "SINGLE_DOUBLE" CACHE STRING "Lammps gpu precision size")
     set_property(CACHE GPU_PREC PROPERTY STRINGS SINGLE_DOUBLE SINGLE_SINGLE DOUBLE_DOUBLE)
-    add_definitions(-D_${GPU_PREC})
-    add_definitions(-DNV_KERNEL -DUCL_CUDADR)
     option(CUDPP_OPT "Enable CUDPP_OPT" ON)
-
-    set(GPU_SOURCES_DIR ${LAMMPS_SOURCE_DIR}/GPU)
-    set(GPU_SOURCES ${GPU_SOURCES_DIR}/gpu_extra.h)
-
-    set_property(GLOBAL PROPERTY "GPU_SOURCES" "${GPU_SOURCES}")
-
-    # detects styles which have GPU version
-    RegisterStylesExt(${GPU_SOURCES_DIR} opt GPU_SOURCES)
-
-    get_property(GPU_SOURCES GLOBAL PROPERTY GPU_SOURCES)
     
     file(GLOB GPU_LIB_SOURCES ${LAMMPS_LIB_SOURCE_DIR}/gpu/*.cpp)
     file(GLOB GPU_LIB_CU ${LAMMPS_LIB_SOURCE_DIR}/gpu/*.cu ${CMAKE_SOURCE_DIR}/gpu/*.cu)
     file(GLOB_RECURSE GPU_NOT_LIB_CU ${LAMMPS_LIB_SOURCE_DIR}/gpu/lal_pppm.cu)
     list(REMOVE_ITEM GPU_LIB_CU ${GPU_NOT_LIB_CU})
-    include_directories(${GPU_SOURCES_DIR} ${LAMMPS_LIB_SOURCE_DIR}/gpu ${LAMMPS_LIB_BINARY_DIR}/gpu)
+
+    # still missing a way to limit defines to GPU build only
+    # something like cuda_compile_definitions()
+    # one way is this, the other adding it to OPTIONS
+    add_definitions(-D_${GPU_PREC})
+    add_definitions(-DNV_KERNEL -DUCL_CUDADR)
```

Comment:
Aren't `NV_KERNEL` and `UCL_CUDADR` only use in `lib/gpu`? So we should make it PRIVATE defines.

Reply:
it's needed during the `cuda_compile`. As I mentioned in the comment, I didn't find a way to limit it to this.

Reply:
I will have a look.

Reply:
I think I would rewrite this to use `OPTIONS`. Any objections?

Reply:
Since in that case we know it's `nvcc` we can use compiler specific flags, so yeah, ok.